### PR TITLE
New Weapon - WT-650

### PR DESCRIPTION
**Pull request by MajManatee**
# Document the changes in your pull request

The WT-650 Assault rifle. Chambered in 5.45, it's a hold-over from before laser weapons were an option, and still favored by many. It was designed in conjunction with the WT-550 rifle as its bigger, harder-hitting brother.

A variant of the gun will be given to the peacekeeper ERT with rubber rounds, as well as another with laser rounds and rechargeable mags given to the Amber Taskforce. 
The normal rifle will be buyable from cargo in batches of two.
Ammo needs to be researched alongside the WT-550 ammo, and more can also be ordered from cargo.

# Spriting
Waiting on @KillerOrcaCora for the main weapon sprite.
NEARLY done. Final touch-ups in progress.

The rifle
image

Standard mag
image

Standard round
image

Rubber mag
image

Rubber round
image

Incendiary mag
image

Incendiary round
image

Armor-Piercing mag
image

Armor-Piercing round
image

Cased-Laser mag
image

# Wiki Documentation

Changes to Peacekeeper equipment
Changes to Buyable crates from cargo
New weapon and related information.

# Changelog

:cl:  
wip: This is work in progress 
rscadd: Adds a new rifle, the WT-650 Assault Rifle.
tweak: gives the ATF a more reasonable firearm
imageadd: Sprites for new weapon, magazines, and bullets.
/:cl:

### CODE REVIEW

**Comment by ynot01** [comment]
please just make a printable ballistic weapon

**Comment by MajManatee** [comment]
https://github.com/yogstation13/Yogstation/pull/14444 
that is handled here

**Comment by Marmio64** [comment]
1. peacekeeper ERT is supposed to have weak, nonlethal weapons, why do they need a full on assault rifle? If you meant the Amber Task Force are supposed to get this, they dont need it either, Nanotrasen uses energy weaponry.
2. Just seems unnecessary for another ballistic weapon when Nanotrasen is supposedly mostly about the energy weapons. We have shotguns (heavy hitter) and the WT-550 (the light hitter), best to leave the rifles for the syndicate to have a more recognizable design.

**Comment by MajManatee** [comment]
please see the latest push where the peacekeepers get a rubber mag version.

**Comment by cowbot92** [comment]
I actually like the idea, could work as long as it stays rather well balanced.

**Comment by Marmio64** [comment]
> please see the latest push where the peacekeepers get a rubber mag version.

why do they need an big assault rifle with rubber bullets when they already have a smaller one with rubber bullets, which is intended because they're supposed to be a weak ERT. (You send in a normal ERT if the situations dangerous enough to require it, or amber task force)

**Comment by maxion12345** [comment]
> 1. peacekeeper ERT is supposed to have weak, nonlethal weapons, why do they need a full on assault rifle? If you meant the Amber Task Force are supposed to get this, they dont need it either, Nanotrasen uses energy weaponry.
> 2. Just seems unnecessary for another ballistic weapon when Nanotrasen is supposedly mostly about the energy weapons. We have shotguns (heavy hitter) and the WT-550 (the light hitter), best to leave the rifles for the syndicate to have a more recognizable design.

if you kill the people disturbing the peace, you'll have peace

**Comment by Hopekz** [comment]
Cool weapon, weird implementation

**Comment by MajManatee** [comment]
As of this post, Most everything is done aside from the weapon sprite itself.

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 1]
```diff
@@ -127,6 +127,30 @@
...
+	can_bayonet = TRUE
+	knife_x_offset = 27
+	knife_y_offset = 12
+	mag_display = TRUE
+	mag_display_ammo = TRUE
+	weapon_weight = WEAPON_HEAVY
+
+/obj/item/gun/ballistic/automatic/wt650/peacekeeper // So peacekeepers can start with a rubber mag instead of a lethal one.
+	mag_type = /obj/item/ammo_box/magazine/wt650/rubber
+	weapon_weight = WEAPON_HEAVY
```
Comment (SomeguyManperson) [inline code/modules/projectiles/guns/ballistic/automatic.dm:149]:
```suggestion
```

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 2]
```diff
@@ -127,6 +127,30 @@
...
+
+/obj/item/gun/ballistic/automatic/wt650/amber // So ATF members can have laser ammo because idk its cool.
+	mag_type = /obj/item/ammo_box/magazine/wt650/laser
+	weapon_weight = WEAPON_HEAVY
```
Comment (SomeguyManperson) [inline code/modules/projectiles/guns/ballistic/automatic.dm:153]:
```suggestion
```

File: code/modules/projectiles/projectile/bullets/rifle.dm [hunk 3]
```diff
@@ -5,6 +5,28 @@
 	damage = 35
 	wound_bonus = -40
 
+// 5.45 (WT-650 Assault Rifle)
+
+/obj/item/projectile/bullet/a545
+	name = "5.45mm bullet"
+	damage = 30
+	wound_bonus = -40
```
Comment (SomeguyManperson) [inline code/modules/projectiles/projectile/bullets/rifle.dm:13]: the other bullets don't inherit this because they aren't subtypes of /a545

**Comment by SomeguyManperson** [comment]
why

**Comment by TheRyeGuyWhoWillNowDie** [comment]
giving these to ATF is kind of a nerf since with the laser rifle you can take the packs out and recharge them, while the weapon itself is still EMP-proof, you cant do that with these.

**Comment by MusselMain** [comment]
> 1. peacekeeper ERT is supposed to have weak, nonlethal weapons, why do they need a full on assault rifle? If you meant the Amber Task Force are supposed to get this, they dont need it either, Nanotrasen uses energy weaponry.
> 2. Just seems unnecessary for another ballistic weapon when Nanotrasen is supposedly mostly about the energy weapons. We have shotguns (heavy hitter) and the WT-550 (the light hitter), best to leave the rifles for the syndicate to have a more recognizable design.

Have you seen how the Amber Task Force's main rifle functions? it's basically a laser rifle that is pretty much the exact same as it's described in this PR, it's a weaker damage 20+1 shot laser rifle that uses magazines that hold "battery" bullets and can be recharged at any recharger and the Amber Task Force is described as NT's equivalent to the Space Marines, meaning it'd make sense for them to have access to a rifle that's chosen for it's ease of finding ammo, and I honestly agree with this part at least, because the current Laser Rifle's mags are super rare, and giving a ERT that's designed for fighting large scale battles a weapon that can use a variety of ammo types that are easily accessible makes a lot of sense, (especially if we're assuming this is a military force that'll need all the ammo they can find). 

Also while NT uses a lot of Energy and laser weapons, they're not afraid to use ballistics weapons as well, just look at the: double barrel (either issued to or brought by bartender), combat shotgun (ordered in cargo, and is a NT gun), riot shotgun (Security), WT-550 (Older SMG/Carbine cargo/security), Compact Combat Shotgun (Issued to or made by Warden), Detective's Revolver  (issued to or brought by Detective), Unica 6 Auto revolver (Issued to CENTCOMM officers), NT-ARG "Boarder" (another Assault rifle, issued to NT boarders), NT SABR SMG (can use suppressors, presumably used by NT Assassin style ops). All to say they're not afraid to use ballistics weapons, especially if said ballistic weapons are a cheaper alternative to laser guns, while holding the versatility of a different ammo types (which is what they're trying to push here, and I like it considering it fits with NT and many other real life countries doctrine of "if it still works, why replace it?"). 

I also believe this has kind of been needed for a while, NT has always proven the have the tech and RD invested in Laser weapons/Energy weapons, but I also think they've never shown a NT issued rifle (that isn't the insanely strong admin spawn NT-ARG) on Yog, and think it'd fit the niche very well if it's balanced correctly. I will say, I'd rather not have the old Laser Rifle be completely removed, as It's held a dear place in my heart (especially the times where I found it on a space ruin) and I hate removing features from the game, however I do think this will be a really cool addition to NT's arsenal.

I'd recommend making this new rifle close to/more expensive then the Combat shotgun (if it ends up being that effective), so that it's the sort of weapon that'll cost you a fair amount to equip a few people with a great weapon, and still leaving the cheaper WT-550 a spot as a cheaper, easily accessible carbine.

**Comment by Hopekz** [comment]
> you can take the packs out and recharge them

To be fair that's a bug that I didn't feel like fixing when I designed and made amber, not an actual "feature"
I thought it was funny that physical bullets were created out of thin air from a recharger and didn't think people would even find it as a feature of the weapon since it didn't make sense.

**Comment by stwordcitizen** [comment]
30 damage bullets are really strong maybe 25-27 would be better?

File: code/modules/projectiles/guns/ballistic/automatic.dm (hunk 2 above)
Comment (MajManatee) [reply code/modules/projectiles/guns/ballistic/automatic.dm:153]: Without it there, same with the other, weapons were not having a weight set.

**Comment by MajManatee** [comment]
The ATF get special rifles and mags that can be recharged, the same as the old rifle they had. As of now, these magazines are not otherwise obtainable.

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 4]
```diff
@@ -127,6 +127,30 @@
 	mag_display = TRUE
 	empty_indicator = TRUE
 
+/obj/item/gun/ballistic/automatic/wt650
+	name = "WT-650 Assault Rifle"
+	desc = "A long-barreled, assault variant of the WT-550. Uses 5.56 rounds and is specially designed for assault teams, PMC forces, and armies favouring balistic weapons over their laser counterparts."
```
Comment (Skrem7) [inline code/modules/projectiles/guns/ballistic/automatic.dm]:
```suggestion
	desc = "A long-barreled, assault variant of the WT-550. Uses 5.45 rounds and is specially designed for assault teams, PMC forces, and armies favouring ballistic weapons over their laser counterparts."
```

File: code/modules/projectiles/boxes_magazines/external/rifle.dm [hunk 5]
```diff
@@ -20,3 +20,77 @@
...
+/obj/item/ammo_box/magazine/wt650/incendiary/update_icon()
+	..()
+	if(ammo_count())
+		icon_state = "wt650_fire"
+	else
+		icon_state = "wt650_fire_empty"
+		
+/obj/item/ammo_box/magazine/wt650/laser
+	name = "WT-650 laser magazine (5.45mm)."
+	desc = "A 5.45mm magazine for the WT-650. This one is specially designed to print more ammo when put into any standard weapon recharger. Convienent!"
```
Comment (Skrem7) [inline code/modules/projectiles/boxes_magazines/external/rifle.dm]:
```suggestion
	desc = "A 5.45mm magazine for the WT-650. This one is specially designed to print more ammo when put into any standard weapon recharger. Convenient!"
```

File: code/modules/research/designs/weapon_designs.dm [hunk 6]
```diff
@@ -411,3 +411,43 @@
...
+	id = "mag_wt650"
+	build_type = PROTOLATHE
+	materials = list(/datum/material/iron = 5000)
+	build_path = /obj/item/ammo_box/magazine/wt650
+	category = list("Ammo")
+	departmental_flags = DEPARTMENTAL_FLAG_ARMORY
+
+/datum/design/mag_wt650/rubber
+	name = "WT-650 Assault Rifle Rubber Magazine (5.45mm)"
+	desc = "A 30 round non-lethal rubber magazine for the WT-650 Assault Rifle."
```
Comment (Skrem7) [inline code/modules/research/designs/weapon_designs.dm]:
```suggestion
	desc = "A 30 round less-lethal rubber magazine for the WT-650 Assault Rifle."
```

File: code/modules/projectiles/projectile/bullets/rifle.dm [hunk 7]
```diff
@@ -5,6 +5,28 @@
...
+
+/obj/item/projectile/bullet/a545_rubber
+	name = "5.45mm rubber bullet"
+	damage = 5
+	stamina = 25
```
Comment (Skrem7) [inline code/modules/projectiles/projectile/bullets/rifle.dm]:
```suggestion
	stamina = 45
```

File: code/modules/projectiles/projectile/bullets/rifle.dm [hunk 8]
```diff
@@ -5,6 +5,28 @@
...
+
+/obj/item/projectile/bullet/a545_ap
+	name = "5.45mm armor-piercing bullet"
+	damage = 18
+	armour_penetration = 50
```
Comment (Skrem7) [inline code/modules/projectiles/projectile/bullets/rifle.dm]:
```suggestion
	damage = 27
	armour_penetration = 40
```

File: code/modules/projectiles/projectile/bullets/rifle.dm [hunk 9]
```diff
@@ -5,6 +5,28 @@
...
+	
+/obj/item/projectile/bullet/incendiary/a545_incendiary
+	name = "5.45mm incendiary bullet"
+	damage = 12
+	fire_stacks = 1
```
Comment (Skrem7) [inline code/modules/projectiles/projectile/bullets/rifle.dm]:
```suggestion
	damage = 15
	fire_stacks = 2
```

**Review by Skrem7 — COMMENTED** [review]
MAkes alternative ammos to be relevant relative to the WT. Essentially this is just a much more ammo-efficient WT-550 as it currently stands I would 100% recommend adding HP rounds for shits and giggles.

**Comment by JamieD1** [comment]
Whats the reason for giving security a 546 carbine? I don't understand why this is even needed.

File: code/modules/cargo/packs.dm [hunk 10]
```diff
@@ -703,6 +703,38 @@
 	contains = list(/obj/item/ammo_box/magazine/wt550m9)
 	small_item = TRUE
 
+/datum/supply_pack/security/armory/wt650
+	name = "WT-650 Assualt Rifle Crate"
+	desc = "Contains two high-powered, semiautomatic rifles chambered in 5.45mm rounds. Requires Armory access to open."
```
Comment (Anvilman6) [inline code/modules/cargo/packs.dm]:
```suggestion
	desc = "Contains two high-powered semiautomatic rifles, chambered in 5.45mm rounds. Requires Armory access to open."
```

File: code/modules/cargo/packs.dm [hunk 11]
```diff
@@ -703,6 +703,38 @@
...
+	cost = 12500
+	contains = list(/obj/item/gun/ballistic/automatic/wt650,
+					/obj/item/gun/ballistic/automatic/wt650)
+	crate_name = "assault rifle crate"
+
+/datum/supply_pack/security/armory/wt550_single
+	name = "WT-650 Assualt Rifle Single-Pack"
+	desc = "Contains one high-powered, semiautomatic rifle chambered in 5.45mm rounds. Requires Armory access to open."
```
Comment (Anvilman6) [inline code/modules/cargo/packs.dm]:
```suggestion
	desc = "Contains one high-powered semiautomatic rifle, chambered in 5.45mm rounds. Requires Armory access to open."
```

File: code/modules/cargo/packs.dm [hunk 12]
```diff
@@ -703,6 +703,38 @@
...
+	cost = 8000
+	contains = list(/obj/item/gun/ballistic/automatic/wt650)
+	small_item = TRUE
+
+/datum/supply_pack/security/armory/wt550ammo
+	name = "WT-650 Assualt Rifle Ammo Crate"
+	desc = "Contains four 30-round magazines for the WT-650 Assualt Rifle. Each magazine is designed to facilitate rapid tactical reloads. Requires Armory access to open."
```
Comment (Anvilman6) [inline code/modules/cargo/packs.dm]:
```suggestion
	desc = "Contains four 30-round magazines for the WT-650 Assault Rifle. Requires Armory access to open."
```
Possibly unneeded word bloat, unless there's an actual mechanical basis for this that I missed. Remove or keep at your discretion.

File: code/modules/cargo/packs.dm [hunk 13]
```diff
@@ -703,6 +703,38 @@
...
+	cost = 4500
+	contains = list(/obj/item/ammo_box/magazine/wt650,
+					/obj/item/ammo_box/magazine/wt650,
+					/obj/item/ammo_box/magazine/wt650,
+					/obj/item/ammo_box/magazine/wt650)
+
+/datum/supply_pack/security/armory/wt550ammo
+	name = "WT-650 Assualt Rifle Ammo Crate"
+	desc = "Contains four less than lethal 30-round magazines for the WT-650 Assualt Rifle. Each magazine is designed to facilitate rapid tactical reloads. Requires Armory access to open."
```
Comment (Anvilman6) [inline code/modules/cargo/packs.dm]:
```suggestion
	desc = "Contains four 30-round rubber shot magazines for the WT-650 Assault Rifle. Requires Armory access to open."
```
See above on word bloat

File: code/modules/projectiles/boxes_magazines/external/rifle.dm [hunk 14]
```diff
@@ -20,3 +20,77 @@
...
+/obj/item/ammo_box/magazine/wt650/incendiary/update_icon()
+	..()
+	if(ammo_count())
+		icon_state = "wt650_fire"
+	else
+		icon_state = "wt650_fire_empty"
+		
+/obj/item/ammo_box/magazine/wt650/laser
+	name = "WT-650 laser magazine (5.45mm)."
+	desc = "A 5.45mm magazine for the WT-650. This one is specially designed to print more ammo when put into any standard weapon recharger. Convenient!"
```
Comment (Anvilman6) [inline code/modules/projectiles/boxes_magazines/external/rifle.dm]:
```suggestion
	desc = "A 5.45mm magazine for the WT-650. This one will print more ammo when plugged into any standard weapon recharger. Convenient!"
```

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 15]
```diff
@@ -127,6 +127,30 @@
 	mag_display = TRUE
 	empty_indicator = TRUE
 
+/obj/item/gun/ballistic/automatic/wt650
+	name = "WT-650 Assault Rifle"
+	desc = "A long-barreled, assault variant of the WT-550. Uses 5.45 rounds and is specially designed for assault teams, PMC forces, and armies favouring ballistic weapons over their laser counterparts."
```
Comment (Anvilman6) [inline code/modules/projectiles/guns/ballistic/automatic.dm]:
```suggestion
	desc = "A long-barreled assault variant of the WT-550. Uses 5.45mm rounds. and is specially for heavy ballistic warfare."
```

File: code/modules/research/designs/weapon_designs.dm [hunk 16]
```diff
@@ -411,3 +411,43 @@
...
+	id = "mag_wt650"
+	build_type = PROTOLATHE
+	materials = list(/datum/material/iron = 5000)
+	build_path = /obj/item/ammo_box/magazine/wt650
+	category = list("Ammo")
+	departmental_flags = DEPARTMENTAL_FLAG_ARMORY
+
+/datum/design/mag_wt650/rubber
+	name = "WT-650 Assault Rifle Rubber Magazine (5.45mm)"
+	desc = "A 30 round less-lethal rubber magazine for the WT-650 Assault Rifle."
```
Comment (Anvilman6) [inline code/modules/research/designs/weapon_designs.dm]:
```suggestion
	desc = "A 30 round rubber shot magazine for the WT-650 Assault Rifle."
```

**Review by Anvilman6 — CHANGES_REQUESTED** [review]
Initial review

File: code/modules/cargo/packs.dm (hunk 12 above)
Comment (MajManatee) [reply code/modules/cargo/packs.dm]: it IS a function in game, but I don't care about the description

**Comment by KillerOrcaCora** [comment]
Leaving this open for anyone to pick up will close in a month, sprites are fully done nothing but corn flakes and most likely balance changes left. (And of course adding the new sprites)

**Comment by Mqiib** [comment]
image

This is what is suggested

**Comment by JamieD1** [comment]
> image
> 
> This is what is suggested

^

**Comment by Hopekz** [comment]
> > image
> > 
> > This is what is suggested
> 
> ^

I remember a time where simple changes like this were just pushed through by maintainers and HC's.

If it's a requirement and you're going to be waiting weeks on it why don't you just hit the edit button and change the values then merge

**Comment by 00ze-cyclone** [comment]
Someone give me a situation for sec where this gun is a better option than laser, wt550 or a shotgun.
Currently the only thing that gun has going for it is an absurdly big magazine. And combat in ss13 tend to be quick, you'll rarely need more than 20 shots for an engagement and then you can go recharge or get more bullets.
That gun keep looking more and more like pointless bloat just there because "ballistic cool".

**Comment by Skrem7** [comment]
> 

Quite literally just WT-550 values but merged into a single bullet, no AP on the base bullet nor the rubber round. This is much less compact than the WT and more expensive, with only the upside of a massive magazine (which is immensely notable, to be fair).

**Comment by JamieD1** [comment]
> > image
> > This is what is suggested
> 
> ^

DM when you do this and solve conflicts.

**Comment by 00ze-cyclone** [comment]
> Quite literally just WT-550 values but merged into a single bullet, no AP on the base bullet nor the rubber round. This is much less compact than the WT and more expensive, with only the upside of a massive magazine (which is immensely notable, to be fair).

The trade off of more ammo per magazine for increased price and no AP is not worth it in the slightest. Ballistics weapons already have tactical reload meaning a small magazine is way less of an issue for them than for energy weapons, and this is trading off the ability to actually hurt things for that not very needed ammo capacity.

The purpose of this weapon seem to fight for prolonged periods of time without resupplying, there’s really only one situations where it happen, blood cult. And WT650 utterly suck against blood cult compared to the WT550 because you’ll be facing armies of cultists with shielded armors, in that situation the burst fire of the WT550 mean you break the forcefield of the cultists way faster and the AP will let you deal more damage, meaning you might start to hurt them before they have already reached melee range to rip you apart with their blade. Having a big magazine is useless if you die at the first engagement.

Nuke ops are also where WT550 is widely used and it also is a way better option there than WT650, for pretty much the same reasons (break forcefields and go through armor) and there the ammo capacity matter even less because nuke ops will remain on your screen for 5 seconds before one of you two die due to the nuke ops immense firepower. You’ll die, run away or defeat the nukie way before you need to reload and then you can do it without pressure, completely nullifying the issue that you can’t shoot during the 2 seconds you have to reload.

Every other antag just use shotgun or laser because they are either weak to lasers or will die way before you emptied the tube of your shotgun.

Oh and also apparently we want to give it to the ATF and peacekeeper ERT because they need a better weapon (despite that gun being pretty much garbage). Nobody seem to have realized that we gave the peacekeeper zipties instead of handcuffs and the ATF no sunglasses for a reason. Those guys are supposed to be weak, if an admin want a better fighting force they send code blue or red.

All of this stem from the fact that the WT650 bring absolutely nothing new to the table, it does everything the WT550 does, it has the same purpose. So it’s either WT+ and it’s OP or it’s expensive sidegrade and is worthless, currently it’s a sidegrade but that decided to trade what matter the most (the ability to actually hurt things) with what matter the least (magazine size), making it even more worthless.

I thought it was obvious but apparently no.

**Comment by Skrem7** [comment]
If the people who said they are going to touch this are not then I'm going to snatch it to make it an ERT DMR that fires a higher caliber, just so the sprites can actually be used.

**Comment by Mqiib** [comment]
> If the people who said they are going to touch this are not then I'm going to snatch it to make it an ERT DMR that fires a higher caliber, just so the sprites can actually be used.

y e s

**Comment by Bibby0110** [comment]
Stale + see above

**Pull request closed without merging**